app/com/linkedin/drelephant/tuning/AbstractAlgorithmManager.java

```diff
@@ -0,0 +1,44 @@
+package com.linkedin.drelephant.tuning;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Logger;
+
+
+public abstract class AbstractAlgorithmManager implements Manager {
+  protected final String JSON_CURRENT_POPULATION_KEY = "current_population";
```

Comment:
As the generation of param for each job is different, we can have a thread pool here, in the baseline manager, and in fitness compute manager as well to parallelize execution as much as possible.

Reply:
Yes we are planning to have ThreadPool for each manager . That would be in nxt version